pre_commit/languages/rust.py

```diff
@@ -52,15 +93,57 @@ def _add_dependencies(
         f.truncate()
 
 
+def install_rust_with_toolchain(toolchain: str) -> None:
+    # Check if rustup is already present in PATH
+    rustup_exe = parse_shebang.find_executable('rustup')
+    with tempfile.TemporaryDirectory() as rustup_dir:
+        rustup_env = (
+            ('RUSTUP_HOME', rustup_dir),
+        )
+        with envcontext(rustup_env):
+            if rustup_exe is None:
+                # We did not detect rustup and need to download it first.
+                if sys.platform == 'win32':
+                    if platform.machine() == 'x86_64':
+                        url = 'https://win.rustup.rs/x86_64'
+                    else:
+                        url = 'https://win.rustup.rs/i686'
+                else:
+                    url = 'https://sh.rustup.rs'
+
+                rustup_init_exe = os.path.join(
+                    rustup_dir,
+                    win_exe('rustup-init'),
+                )
+                urllib.request.urlretrieve(url, rustup_init_exe)
+                make_executable(rustup_init_exe)
+                _, _, _ = cmd_output_b(
+                    rustup_init_exe, '-y', '--quiet', '--no-modify-path',
+                    '--default-toolchain', 'none',
+                    retcode=0,
+                )
+
+                # Now, rustup should be present in `$CARGO_HOME/bin`.
+                rustup_exe = parse_shebang.find_executable('rustup')
+
+            if rustup_exe is None:
+                raise AssertionError(
+                    'failed to find rustup even though it should have been '
+                    'installed at this point -- this is probably a bug',
+                )
+
+            _, _, _ = cmd_output_b(
+                rustup_exe, 'toolchain', 'install', '--no-self-update',
+                toolchain, retcode=0,
+            )
+
+
 def install_environment(
         prefix: Prefix,
-        version: str,
+        language_version: str,
```

Comment:
ah -- yeah I think the better fix is to rename the `version` below to not clobber this (since all the languages use this argument name, best to keep the consistency here)

Reply:
https://github.com/pre-commit/pre-commit/pull/2534/commits/c81a7e2830240cae6627e46dc203f1e686b31582